src/Cryptol/TypeCheck/TypePat.hs

```diff
@@ -193,4 +195,26 @@ anError k = \a -> case tNoUser a of
                     _                                     -> mzero
 
 
-
+{-
+Note [-Wincomplete-uni-patterns and irrefutable patterns]
```

Comment:
@yav: Let me know if you're happy with this approach to `-Wincomplete-uni-patterns` + irrefutable patterns.